### PR TITLE
#93 #83 84

### PR DESCRIPTION
**Pull request by charl26**

### CODE REVIEW

File: BE/MedibillTransactService/MedibillTransactService/Database/PatientAccessClass.cs [hunk 1]
```diff
@@ -10,9 +10,13 @@ namespace MedibillTransactService
 {
```
Comment (DJRave) [inline BE/MedibillTransactService/MedibillTransactService/Database/PatientAccessClass.cs]: Please join you pull requests as I don't like see the same changes 3 times if you want to make 3 PR then join them in to the branch that it needs to go into else you can really cause some very bad merge conflicts

File: src/app/_components/schedules/schedules.component.ts [hunk 2]
```diff
@@ -137,19 +141,31 @@ export class SchedulesComponent implements OnInit {
...
-  onSearchPatient() {
-    if (this.SearchPatientForm.invalid) {
-      return;
-    }
-    this.PatientStore.dispatch(
-      new ViewPatientByCriteria({
-        SearchCriteria: this.SearchPatientForm.value.SearchCriteria,
-        CriteriaValue: this.SearchPatientForm.value.CriteriaValue
+  Select(element: ScheduleItem) {
+    console.log(element);
```
Comment (DJRave) [inline src/app/_components/schedules/schedules.component.ts]: remove the log

File: src/app/_components/transactions/transactions.component.html [hunk 3]
```diff
@@ -1,56 +1,108 @@
...
 
-<mat-accordion *ngIf="PatientInfoSource">
-  <mat-expansion-panel>
-    <mat-expansion-panel-header>
-      <mat-panel-title>
-        PatientInfo
-      </mat-panel-title>
-      <div>
-        <table [dataSource]="PatientInfoSource" mat-table>
+<div *ngIf="patient !== undefined">
```
Comment (DJRave) [inline src/app/_components/transactions/transactions.component.html]: not needed use the javascript truth table just *ngIf="patient" else are you saying the if the patient = null it is correct

File: src/app/_components/transactions/transactions.component.html [hunk 4]
```diff
@@ -134,7 +199,7 @@ <h1>Transactions</h1>
 
     <ng-container matColumnDef="Date">
       <th *matHeaderCellDef mat-header-cell>Procedure Date</th>
-      <td *matCellDef="let element" mat-cell> {{element.ProcedureDate}} </td>
+      <td *matCellDef="let element" mat-cell> {{element.ProcedureDate | date: 'dd/MM/yyyy'}} </td>
```
Comment (DJRave) [inline src/app/_components/transactions/transactions.component.html]: Please make sure the ProcedureDate has a value as the date pipe can give a very mean error

File: src/app/_components/transactions/transactions.component.ts [hunk 5]
```diff
@@ -86,19 +139,38 @@ export class TransactionsComponent implements OnInit {
...
+
+    this.TransactionStore.dispatch(new AddTransaction({
+      ICD10Code: this.GeneralInformation.ICD10Code.value,
+      procedureDetails: this.dataSource.data,
+      AuthorisationCode: this.addPatientInfo.AuthorisationCode.value,
+      LocationID: this.addPatientInfo.Location.value,
+      BMI: this.GeneralInformation.BMI.value,
+      Weight: this.GeneralInformation.Weight.value,
+      Height: this.GeneralInformation.Height.value,
+      ClaimDate: formatDate(this.addPatientInfo.ClaimDate.value, 'dd/MM/yyyy', 'en-US'),
```
Comment (DJRave) [inline src/app/_components/transactions/transactions.component.ts]: Why are you using english US time format

File: src/app/_services/auth.service.ts [hunk 6]
```diff
@@ -46,6 +46,9 @@ export class AuthService {
   }
   updateTokenAndSession() {}
   logout() {
+    localStorage.clear();
```
Comment (DJRave) [inline src/app/_services/auth.service.ts]: Better to add this logic with the effect and not with the call try to keep all the logic together

File: src/app/_components/transactions/transactions.component.ts (hunk 5 above)
Comment (charl26) [reply src/app/_components/transactions/transactions.component.ts]: US time format seems to work with the DateTime in the BE

File: src/app/_services/auth.service.ts (hunk 6 above)
Comment (charl26) [reply src/app/_services/auth.service.ts]: will do

**Pull request merged**